nixos/modules/services/web-apps/openwebrx.nix

```
@@ -0,0 +1,55 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.openwebrx;
+in
+{
+  options.services.openwebrx = with lib; {
+    enable = mkEnableOption "Enable OpenWebRX Web interface for Software-Defined Radios";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.openwebrx;
+    };
+
+    user = mkOption {
+      type = types.str;
+      default = "openwebrx";
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "openwebrx";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    users = {
+      users.${cfg.user} = {
+        isSystemUser = true;
+        group = cfg.group;
+        home = "/var/lib/openwebrx";
+      };
+      groups.${cfg.group} = {};
+    };
+
+    systemd.services.openwebrx = {
+      wantedBy = [ "multi-user.target" ];
+      path = with pkgs; [
+        csdr
+        alsaUtils
+        netcat
+      ];
+      serviceConfig = {
+        ExecStart = "${cfg.package}/bin/openwebrx";
+        Restart = "always";
+        User = cfg.user;
+        Groups = cfg.group;
+        WorkingDirectory = config.users.users.${cfg.user}.home;
+      };
+    };
+
+    systemd.tmpfiles.rules = [
+      "d ${config.users.users.${cfg.user}.home} 0755 ${cfg.user} ${cfg.group} -"
+    ];
```

```suggestion
    systemd.services.openwebrx.serviceConfig.StateDirectory = "openwebrx";
```

[StateDirectory](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#RuntimeDirectory=) is superior to `systemd.tmpfiles.rules` for a number of reasons.